### ISSUE TITLE
Simaltaneous File Uploads when set to multiple

### ISSUE THREAD

**Issue by abishekrsrikaanth** [lian-yue/vue-upload-component#44]
So far this has been working great with vuex-2 and its configurations and examples has helped me so much. Thank you. Is there a way to enable simultaneous file uploads instead of one after the other?

**Comment by lian-yue** [comment]
This feature has been added

**Comment by abishekrsrikaanth** [comment]
@lian-yue, tried to use the new `thread` prop as you have shown on the docs and the example, but doesn't seem to be working. Am i missing any other config other than `thread`? Look forward to your reply.

**Comment by lian-yue** [comment]
Html4 does not support `thread '
Please try reinstalling

```bash
npm uninstall --save vue-upload-component
npm install --save vue-upload-component@next
```


```jsx
<file-upload  :thread="4"></file-upload>
```

**Comment by abishekrsrikaanth** [comment]
@lian-yue, Yes that worked. Thank you very much. I have 3 other questions, not sure if I should open them as separate issues for better clarity.

1. Is there a way to retry or a method that may allow retrying a download if a file failed.
2. Is there a property that may indicate that a file failed to upload
3. Is there an event that is triggered if a file fails to upload, something like an `onError`

**Comment by lian-yue** [comment]
```jsx
<file-upload  :events="events"></file-upload>

```


```js
events: {
  after(file, component) {
    if (file.error) {
      /*onError()*/
      var retry = file.retry || 0
      retry++
      if (retry < 3) {
        component.addFileUpload({retry: retry, file:file.file, size:file.size, name: file.name, el: file.el})
      }
    } else if (file.success) {
      /*onSuccess()*/
    }
  },
}
```

**Comment by abishekrsrikaanth** [comment]
This is great. Thank you very much, appreciate your help. I am liking this component more and more as I use it. Closing this issue now.

**Comment by abishekrsrikaanth** [comment]
@lian-yue, this may be a dumb question, but based on the example above:
1. This will add a new file to the queue of existing files, so I will have to copy any data properties that I had previously assigned to the file that is being retried.
2. What triggers the upload after `addFileUpload` has been called? Should I call another method to trigger the upload of the new file?

**Comment by lian-yue** [comment]
1. You can copy `file.data` or use the` add` event to modify` file.data`
2. 
```js
events: {
  add(file, component) {
    if (file.retry && !component.active) {
      component.active = true;
    }
  }
}
```
3. `this.files.indexOf(file)`
4.  It has been fixed
5. `files[0].xhr.status`

**Comment by abishekrsrikaanth** [comment]
@lian-yue Just noticed one other issue with simultaneous uploads. So, let's say I have 10 files to upload and I set the thread to 4. When I initiate the upload, it uploads the first 4, but the next 4 starts only after the completion of the first 4. Is there a way to keep continuity, so if 2 on the first session gets uploaded, then it automatically starts 2 more. At any point there will be 4 simaltaneous uploads instead of 4 at a time. I hope that makes sense. Would you like me to open a new issue for this?
